lib/sles4sap/sdaf_library.pm

```diff
@@ -609,48 +621,97 @@ sub replace_tfvars_variables {
-        '--keyvault', '${key_vault}',    # Deployer key vault containing credentials
-        '--storageaccountname', '${tfstate_storage_account}',    # storage account for tfstate
-        '--subscription', '${ARM_SUBSCRIPTION_ID}',
-        '--tenant_id', '${ARM_TENANT_ID}',
-        '--spn_id', '${ARM_CLIENT_ID}',
-        '--spn_secret', '${ARM_CLIENT_SECRET}',
-        '--auto-approve');    # avoid user interaction
+    my $deploy_command = get_sdaf_deployment_command(
+        deployment_type => $args{deployment_type}, tfvars_filename => $tfvars_filename);
 
     record_info('SDAF exe', "Executing workload zone deployment: $deploy_command");
 
-    my $output_log_file = log_dir() . "/deploy_workload_zone.log";
+    my $output_log_file = log_dir() . "/deploy_$args{deployment_type}.log";
     $deploy_command = log_command_output(command => $deploy_command, log_file => $output_log_file);
-    my $rc = script_run($deploy_command, timeout => 1800);
+    my $rc = script_run($deploy_command, timeout => $args{timeout});
 
-    upload_logs($output_log_file, log_name => 'deploy_workload_zone.log');    # upload logs before failing
+    upload_logs($output_log_file, log_name => $output_log_file);    # upload logs before failing
     die "Workload zone deployment failed with RC: $rc" if $rc;
     record_info('Deploy done');
 }
 
+
+=head2 get_sdaf_deployment_command
+
+    get_sdaf_deployment_command(deployment_type=>$deployment_type, tfvars_filename=>tfvars_filename);
+
+B<deployment_type>: Type of the deployment: workload_zone or sap_system
+
+B<tfvars_filename>: Filename of tfvars file
+
+Function composes SDAF deployment script command for B<sap_system> or B<workload_zone> according to official documentation.
+Although the documentation uses env OS variable references in the command, function replaces them with actual values.
+This is done for better debugging and logging transparency. Only sensitive values are hidden by using references.
+
+=cut
+
+sub get_sdaf_deployment_command {
+    my (%args) = @_;
+    my $cmd;
+
+    if ($args{deployment_type} eq 'workload_zone') {
+        $cmd = join(' ', sdaf_scripts_dir() . '/install_workloadzone.sh',
+            '--parameterfile', $args{tfvars_filename},    # workload zone tfvars file
+            '--deployer_environment', get_os_variable('deployer_env_code'),    # VNET code
+            '--deployer_tfstate_key', get_os_variable('deployerState'),    # tfstate name. State file is stored in storage account.
+            '--keyvault', get_os_variable('key_vault'),    # Deployer key vault containing credentials
+            '--storageaccountname', get_os_variable('tfstate_storage_account'),    # storage account for tfstate
+            '--subscription', get_os_variable('ARM_SUBSCRIPTION_ID'),
+            '--tenant_id', get_os_variable('ARM_TENANT_ID'),
+            '--spn_id', '${ARM_CLIENT_ID}',    # Keep secrets hidden
```

Comment:
How composing a command string with them keep them secret?

Reply:
Oh I mean hidden in serial terminal output.  I'll change the phrasing so it is more clear. 
It refers to the OS variables which are set during az_login().
https://github.com/lpalovsky/os-autoinst-distri-opensuse/blob/154d99958f733e2b5d146c9fab1eead708a611ca/lib/sles4sap/sdaf_library.pm#L191

Setting it like this will result in command showing: 
`somecommand.sh --tenant_id exposed-tenant-id-value  --spn_id ${ARM_CLIENT_ID}`
So `ARM_CLIENT_ID` is shown in output only as a variable, but it is internally interpreted in shell. 

Doe it make sense?